cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -115,14 +115,15 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.MasterC
 }
 
 func removeOldKubeDNSDeploymentIfCoreDNSIsUsed(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
-	if features.Enabled(cfg.FeatureGates, features.CoreDNS) {
+	kubeDNSDeployment, _ := client.AppsV1().Deployments(metav1.NamespaceSystem).Get(kubeadmconstants.KubeDNS, metav1.GetOptions{})
```

Comment:
You need to check the error, otherwise if `kubeDNSDeployment` does not exist, the check below will panic.